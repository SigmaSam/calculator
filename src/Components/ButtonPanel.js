import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = ({ clickHandler }) => {
  const handleClick = (buttonName) => clickHandler(buttonName);
  const group1 = ['AC', '+/-', '%', '/'];
  const group2 = ['7', '8', '9', 'X'];
  const group3 = ['4', '5', '6', '-'];
  const group4 = ['1', '2', '3', '+'];
  const group5 = ['0', '.', '='];
  return (
    <div className="groupContainer">
      <div>
        {group1.map((name, i) => (
          <Button
            key={name}
            color={i === group1.length - 1 ? 'orange' : 'white'}
            name={name}
            clickHandler={handleClick}
          />
        ))}
      </div>
      <div>
        {group2.map((name, i) => (
          <Button
            key={name}
            color={i === group1.length - 1 ? 'orange' : 'white'}
            name={name}
            clickHandler={handleClick}
          />
        ))}
      </div>
      <div>
        {group3.map((name, i) => (
          <Button
            key={name}
            color={i === group1.length - 1 ? 'orange' : 'white'}
            name={name}
            clickHandler={handleClick}
          />
        ))}
      </div>
      <div>
        {group4.map((name, i) => (
          <Button
            key={name}
            color={i === group1.length - 1 ? 'orange' : 'white'}
            name={name}
            clickHandler={handleClick}
          />
        ))}
      </div>
      <div>
        {group5.map((name, i) => (
          <Button
            key={name}
            color={i === group1.length - 1 ? 'orange' : 'white'}
            wide={name === '0'}
            name={name}
            clickHandler={handleClick}
          />
        ))}
      </div>
    </div>
  );
};

export default ButtonPanel;

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};

ButtonPanel.defaultProps = {
  clickHandler: null,
};
