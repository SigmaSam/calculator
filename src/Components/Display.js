import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ result }) => <div className="Display"><h1 className="Result">{result}</h1></div>;

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
