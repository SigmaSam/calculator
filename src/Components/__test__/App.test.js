import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import ButtonPanel from '../ButtonPanel';
import Display from '../Display';

describe('Display Display.js', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(<BrowserRouter><Display /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Display ButtonPanel.js', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(<BrowserRouter><ButtonPanel /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
