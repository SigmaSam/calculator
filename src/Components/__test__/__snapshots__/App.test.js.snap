// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display App.js renders correctly 1`] = `
<div
  className="Display"
>
  <h1
    className="Result"
  >
    0
  </h1>
</div>
`;

exports[`Display ButtonPanel.js renders correctly 1`] = `
<div
  className="groupContainer"
>
  <div>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      /
    </button>
  </div>
  <div>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      X
    </button>
  </div>
  <div>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      -
    </button>
  </div>
  <div>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      +
    </button>
  </div>
  <div>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "50%",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Display Display.js renders correctly 1`] = `
<div
  className="Display"
>
  <h1
    className="Result"
  >
    0
  </h1>
</div>
`;
